keras_cv/layers/detectron2_layers.py

```diff
@@ -0,0 +1,477 @@
+        mlp_dim,
+        num_heads,
+        use_bias=True,
+        use_rel_pos=False,
+        window_size=0,
+        input_size=None,
+        activation="gelu",
+        layer_norm_epsilon=1e-6,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.project_dim = project_dim
+        self.mlp_dim = mlp_dim
+        self.num_heads = num_heads
+        self.use_bias = use_bias
+        self.input_size = input_size
+        self.activation = activation
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.window_size = window_size
+        self.use_rel_pos = use_rel_pos
+
+        self.layer_norm1 = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon
+        )
+        self.layer_norm2 = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon
+        )
+        self.attention = MultiHeadAttentionWithRelativePE(
+            num_heads=self.num_heads,
+            key_dim=self.project_dim // self.num_heads,
+            use_bias=use_bias,
+            use_rel_pos=use_rel_pos,
+            input_size=input_size
+            if window_size == 0
+            else (window_size, window_size),
+        )
+        self.mlp_block = MLP(
+            mlp_dim,
+            project_dim,
+            num_layers=2,
+            activation="gelu",
+        )
+
+    def build(self, input_shape=None):
+        self.layer_norm1.build([None, None, None, self.project_dim])
+        self.layer_norm2.build([None, None, None, self.project_dim])
+        self.attention.build()
+        self.mlp_block.build([None, None, None, self.project_dim])
+        self.built = True
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+    def call(self, x):
+        shortcut = x
+        x = self.layer_norm1(x)
+        # Window Partition
+        if self.window_size > 0:
+            H, W = x.shape[1], x.shape[2]
+
+            x, HW_padded = window_partition(x, self.window_size)
+
+        x = self.attention(x)
+        # Reverse Window Partition
+        if self.window_size > 0:
+            x = window_unpartition(x, self.window_size, HW_padded, (H, W))
+
+        x = shortcut + x
+        x = x + self.mlp_block(self.layer_norm2(x))
+
+        return x
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "project_dim": self.project_dim,
+                "mlp_dim": self.mlp_dim,
+                "num_heads": self.num_heads,
+                "use_bias": self.use_bias,
+                "use_rel_pos": self.use_rel_pos,
+                "window_size": self.window_size,
+                "input_size": self.input_size,
+                "activation": self.activation,
+                "layer_norm_epsilon": self.layer_norm_epsilon,
+            }
+        )
+        return config
+
+
+@keras_cv_export("keras_cv.layers.ViTDetPatchingAndEmbedding")
+class ViTDetPatchingAndEmbedding(keras.layers.Layer):
```

Comment:
This is the same as the ViT patching and embedding but without positional embedding. 
I'm torn between being able to turn off PE in the default layer and adding that as a flag and having a new layer for this...

Reply:
Are you referring to the `PathingAndEmbedding` class for the ViT model in KerasCV? I addressed that here: https://github.com/keras-team/keras-cv/pull/1987#discussion_r1304907301